lit-element.js

```diff
@@ -0,0 +1,193 @@
+import { PropertiesMixin } from '@polymer/polymer/lib/mixins/properties-mixin.js';
+import { camelToDashCase } from '@polymer/polymer/lib/utils/case-map.js';
+import { render } from 'lit-html/lib/shady-render.js';
+export { html } from 'lit-html/lib/lit-extended.js';
+/**
+ * Renders attributes to the given element based on the `attrInfo` object where
+ * boolean values are added/removed as attributes.
+ * @param element Element on which to set attributes.
+ * @param attrInfo Object describing attributes.
+ */
+export function renderAttributes(element, attrInfo) {
+    for (const a in attrInfo) {
+        const v = attrInfo[a] === true ? '' : attrInfo[a];
+        if (v || v === '' || v === 0) {
+            if (element.getAttribute(a) !== v) {
+                element.setAttribute(a, v);
+            }
+        }
+        else if (element.hasAttribute(a)) {
+            element.removeAttribute(a);
+        }
+    }
+}
+/**
+ * Returns a string of css class names formed by taking the properties
+ * in the `classInfo` object and appending the property name to the string of
+ * class names if the property value is truthy.
+ * @param classInfo
+ */
+export function classString(classInfo) {
+    const o = [];
+    for (const name in classInfo) {
+        const v = classInfo[name];
+        if (v) {
+            o.push(name);
+        }
+    }
+    return o.join(' ');
+}
+/**
+ * Returns a css style string formed by taking the properties in the `styleInfo`
+ * object and appending the property name (dash-cased) colon the
+ * property value. Properties are separated by a semi-colon.
+ * @param styleInfo
+ */
+export function styleString(styleInfo) {
+    const o = [];
+    for (const name in styleInfo) {
+        const v = styleInfo[name];
+        if (v || v === 0) {
+            o.push(`${camelToDashCase(name)}: ${v}`);
+        }
+    }
+    return o.join('; ');
+}
+export class LitElement extends PropertiesMixin(HTMLElement) {
+    constructor() {
+        super(...arguments);
+        this.__renderComplete = null;
+        this.__resolveRenderComplete = null;
+        this.__isInvalid = false;
+        this.__isChanging = false;
+    }
+    ready() {
+        this._root = this._createRoot();
+        super.ready();
```

Comment:
The compiled js seems to be out of date. For example I don't see the `this._firstRendered();` here